Content.Client/GameObjects/Components/Weapons/Ranged/Barrels/Visualizers/RangedMagazineVisualizer2D.cs

```diff
@@ -0,0 +1,91 @@
+using Content.Shared.GameObjects.Components.Weapons.Ranged.Barrels;
+using Content.Shared.Utility;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Client.Interfaces.GameObjects.Components;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Utility;
+using YamlDotNet.RepresentationModel;
+
+namespace Content.Client.GameObjects.Components.Weapons.Ranged.Barrels.Visualizers
+{
+    [UsedImplicitly]
+    public sealed class RangedMagazineVisualizer2D : AppearanceVisualizer
```

Comment:
There seems to be a lot of duplication between the various magazine visualisers. Could they not be unified in some way?

Reply:
I could probably combine battery, rangedmagazine and magazinebarrel so I'll give it a shot.